vertx-db2-client/src/main/java/io/vertx/db2client/impl/codec/InitialHandshakeCommandCodec.java

```diff
@@ -54,7 +56,8 @@ void encode(DB2Encoder encoder) {
             //Sometimes DB2 closes the connection when sending an invalid Database name.
             //-4499 = A fatal error occurred that resulted in a disconnect from the data source.
             //08001 = "The connection was unable to be established"
-            cmd.fail(new DB2Exception("Socket closed during connection", -4499,"08001"));           
+            cmd.fail(new DB2Exception("The connection was closed by the database server.", 
+            		SqlCode.CONNECTION_REFUSED, SQLState.CONNECT_UNABLE_TO_CONNECT_TO_SERVER));           
```

Comment:
So here it sounds like we are able to hit the DB2 server, but it closes the connection because of an invalid DB name.  In the comments section of SQLState.java (around line 1737) they make a note that the 08001 errors are for when the client is unable to establish a connection with the server.  Here it seems we are able to connect to the server and they are rejecting the request because of the invalid DB name.  So my suggestion would be to use SQLState.NET_DATABASE_NOT_FOUND instead (which sends back value of "08004*")